src/Form/MiradorConfigForm.php

```diff
@@ -23,6 +30,44 @@ public function getFormId() {
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildForm($form, $form_state);
     $config = $this->config('islandora_mirador.settings');
+    $form['mirador_library_fieldset'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Mirador library location'),
+    ];
+    $form['mirador_library_fieldset']['mirador_library_use_remote'] = [
+      '#type' => 'radios',
+      '#options' => [
+        'local'=> $this->t('Local library placed in /libraries inside your webroot.'),
+        'remote' => $this->t('Remote (e.g. on a CDN)'),
+      ],
+      '#default_value' => $config->get('mirador_library_use_remote'),
+    ];
+
+    $form['mirador_library_fieldset']['mirador_library_location'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Remote Mirador library location'),
+      '#description' => $this->t('Remote URL of compiled Mirador library. It must be minified and be a single file.'),
+      '#default_value' => $config->get('mirador_library_location'),
+      '#states' => [
+        // Show this field only if the 'remote' option is selected above.
+        'enabled' => [
+          ':input[name="mirador_library_use_remote"]' => [
+            'value' => 'remote',
+          ],
+        ],
+      ],
+    ];
```

Comment:
Having this be fully configurable is a substantial security (XSS?) risk, no? Like: If a malicious agent managed to point this at some doctored version of the library, for whatever purposes?

... could be difficult to detect if it still renders the viewer the same, but could make use of legitimate users' sessions to do things behind the scenes.

...  might make more sense as something of a plugin architecture? Like: Have the core module implement the default CDN provider plugin (and `local``/libraries` thing?), but then there could be a "roblib mirador" plugin alternative provided by another module, or whatever other built version alternatives? Then the configuration here becomes selecting between the available alternatives?

Reply:
Is this really an escalation? If you have `administer site configuration` you already have full access to the site, and it'd be much easier to add your weird evil javascript in a formatted text field.

Reply:
Places dealing with "unfiltered" HTML typically have warnings about security implications of their use, which are not present here. Also, creation of content is somewhat auditable (usually writing "created" dates and whatnot; so after their detection, you might be able to identify the window in which things might be questionable). Config changes, not so much.

Use of hooks/plugins allows for this concern to be external to this code.

... Additionally, this is baking in a reference to a particular version of external code that could be better encapsulated in another module... like: If there was a support module wrapping the `mirador-integration-islandora/islandora-mirador-0.1.0.js` business, then, it's up to that other module to maintain the reference, which could be updated just by updating to a newer version of this other hypothetical integration wrapper module.

Reply:
My first thought was to just make the user install the application themselves but it's a several step process to compile the app and configure it to e.g. not be split into chunks by web pack. 

@adam-vessey would you be more comfortable with a setting in settings.php? I want to facilitate local development of Mirador plugin integration but also would like the default install to not be too many steps.

Reply:
FWIW I do not share @adam-vessey 's concern here. But might simply removing the default ameliorate this concern?